packages/celotool/src/lib/geth.ts

```diff
@@ -614,3 +639,501 @@ export const transferERC20Token = async (
+  return path.join(getInstanceDir(runPath, instance), 'snapshot')
+}
+
+export function importGenesis(genesisPath: string) {
+  return JSON.parse(fs.readFileSync(genesisPath).toString())
+}
+
+function getDatadir(runPath: string, instance: GethInstanceConfig) {
+  const dir = path.join(getInstanceDir(runPath, instance), 'datadir')
+  // @ts-ignore
+  fs.mkdirSync(dir, { recursive: true })
+  return dir
+}
+
+/**
+ * @returns Promise<number> the geth pid number
+ */
+export async function initAndStartGeth(
+  gethConfig: GethRunConfig,
+  gethBinaryPath: string,
+  instance: GethInstanceConfig,
+  verbose: boolean
+) {
+  const datadir = getDatadir(gethConfig.runPath, instance)
+
+  if (verbose) {
+    console.info(`geth:${instance.name}: init datadir ${datadir}`)
+  }
+
+  const genesisPath = path.join(gethConfig.runPath, 'genesis.json')
+  await init(gethBinaryPath, datadir, genesisPath, verbose)
+
+  if (instance.privateKey) {
+    await importPrivateKey(gethConfig, gethBinaryPath, instance, verbose)
+  }
+
+  return startGeth(gethConfig, gethBinaryPath, instance, verbose)
+}
+
+export async function init(
+  gethBinaryPath: string,
+  datadir: string,
+  genesisPath: string,
+  verbose: boolean
+) {
+  if (verbose) {
+    console.log(`init geth with genesis at ${genesisPath}`)
+  }
+
+  await spawnCmdWithExitOnFailure('rm', ['-rf', datadir], { silent: !verbose })
+  await spawnCmdWithExitOnFailure(gethBinaryPath, ['--datadir', datadir, 'init', genesisPath], {
+    silent: !verbose,
+  })
+}
+
+export async function importPrivateKey(
+  getConfig: GethRunConfig,
+  gethBinaryPath: string,
+  instance: GethInstanceConfig,
+  verbose: boolean
+) {
+  const keyFile = path.join(getDatadir(getConfig.runPath, instance), 'key.txt')
+
+  fs.writeFileSync(keyFile, instance.privateKey, { flag: 'a' })
+
+  if (verbose) {
+    console.info(`geth:${instance.name}: import account`)
+  }
+
+  const args = [
+    'account',
+    'import',
+    '--datadir',
+    getDatadir(getConfig.runPath, instance),
+    '--password',
+    '/dev/null',
+    keyFile,
+  ]
+
+  if (verbose) {
+    console.log(gethBinaryPath, ...args)
+  }
+
+  await spawnCmdWithExitOnFailure(gethBinaryPath, args, { silent: true })
+}
+
+export async function getEnode(peer: string, ws: boolean = false) {
+  // do we have already an enode?
+  if (peer.toLowerCase().startsWith('enode')) {
+    // yes return peer
+    return peer
+  }
+
+  // no, try to build it
+  const p = ws ? 'ws' : 'http'
+  const enodeRpcUrl = `${p}://localhost:${peer}`
+  const admin = new Admin(enodeRpcUrl)
+
+  let nodeInfo: any = {
+    enode: null,
+  }
+
+  try {
+    nodeInfo = await admin.getNodeInfo()
+  } catch {
+    console.error(`Unable to get node info from ${enodeRpcUrl}`)
+  }
+
+  return nodeInfo.enode
+}
+
+export async function addStaticPeers(datadir: string, peers: string[], verbose: boolean) {
+  const staticPeersPath = `${datadir}/static-nodes.json`
```

Comment:
Like I've mentioned before, it's better to use `path.join` by default to avoid edge cases.
```suggestion
  const staticPeersPath = path.join(datadir, 'static-nodes.json')
```